In [2]:
import pandas as pd
import numpy as np

In [12]:
# read data from the CSV file:
raw_df = pd.read_csv('data/books_1.Best_Books_Ever.csv')

In [15]:
# check number of rows and columns:
df.shape

(52478, 25)

There are 52,478 rows and 25 columns in this dataset

In [13]:
# sanity check the first five rows:
raw_df.head()

bookId  \
0                     2767052-the-hunger-games   
1  2.Harry_Potter_and_the_Order_of_the_Phoenix   
2                   2657.To_Kill_a_Mockingbird   
3                     1885.Pride_and_Prejudice   
4                               41865.Twilight   

                                       title                 series  \
0                           The Hunger Games    The Hunger Games #1   
1  Harry Potter and the Order of the Phoenix        Harry Potter #5   
2                      To Kill a Mockingbird  To Kill a Mockingbird   
3                        Pride and Prejudice                    NaN   
4                                   Twilight   The Twilight Saga #1   

                                      author  rating  \
0                            Suzanne Collins    4.33   
1  J.K. Rowling, Mary GrandPré (Illustrator)    4.50   
2                                 Harper Lee    4.28   
3  Jane Austen, Anna Quindlen (Introduction)    4.26   
4                            Stephenie Meyer    3.60   

                                         description language           isbn  \
0  WINNING MEANS FAME AND FORTUNE.LOSING MEANS CE...  English  9780439023481   
1  There is a door at the end of a silent corrido...  English  9780439358071   
2  The unforgettable novel of a childhood in a sl...  English  9999999999999   
3  Alternate cover edition of ISBN 9780679783268S...  English  9999999999999   
4  About three things I was absolutely positive.\...  English  9780316015844   

                                              genres  \
0  ['Young Adult', 'Fiction', 'Dystopia', 'Fantas...   
1  ['Fantasy', 'Young Adult', 'Fiction', 'Magic',...   
2  ['Classics', 'Fiction', 'Historical Fiction', ...   
3  ['Classics', 'Fiction', 'Romance', 'Historical...   
4  ['Young Adult', 'Fantasy', 'Romance', 'Vampire...   

                                          characters  ... firstPublishDate  \
0  ['Katniss Everdeen', 'Peeta Mellark', 'Cato (H...  ...              NaN   
1  ['Sirius Black', 'Draco Malfoy', 'Ron Weasley'...  ...         06/21/03   
2  ['Scout Finch', 'Atticus Finch', 'Jem Finch', ...  ...         07/11/60   
3  ['Mr. Bennet', 'Mrs. Bennet', 'Jane Bennet', '...  ...         01/28/13   
4  ['Edward Cullen', 'Jacob Black', 'Laurent', 'R...  ...         10/05/05   

                                              awards numRatings  \
0  ['Locus Award Nominee for Best Young Adult Boo...    6376780   
1  ['Bram Stoker Award for Works for Young Reader...    2507623   
2  ['Pulitzer Prize for Fiction (1961)', 'Audie A...    4501075   
3                                                 []    2998241   
4  ['Georgia Peach Book Award (2007)', 'Buxtehude...    4964519   

                                      ratingsByStars likedPercent  \
0  ['3444695', '1921313', '745221', '171994', '93...         96.0   
1  ['1593642', '637516', '222366', '39573', '14526']         98.0   
2  ['2363896', '1333153', '573280', '149952', '80...         95.0   
3  ['1617567', '816659', '373311', '113934', '767...         94.0   
4  ['1751460', '1113682', '1008686', '542017', '5...         78.0   

                                             setting  \
0  ['District 12, Panem', 'Capitol, Panem', 'Pane...   
1  ['Hogwarts School of Witchcraft and Wizardry (...   
2               ['Maycomb, Alabama (United States)']   
3  ['United Kingdom', 'Derbyshire, England (Unite...   
4  ['Forks, Washington (United States)', 'Phoenix...   

                                            coverImg  bbeScore bbeVotes  price  
0  https://i.gr-assets.com/images/S/compressed.ph...   2993816    30516   5.09  
1  https://i.gr-assets.com/images/S/compressed.ph...   2632233    26923   7.38  
2  https://i.gr-assets.com/images/S/compressed.ph...   2269402    23328    NaN  
3  https://i.gr-assets.com/images/S/compressed.ph...   1983116    20452    NaN  
4  https://i.gr-assets.com/images/S/compressed.ph...   1459448    14874    2.1  

[5 rows x 25 columns]